src/tests/end-to-end/playlist.cpp

```diff
@@ -0,0 +1,256 @@
+#define CATCH_CONFIG_MAIN // This tells Catch to provide a main() - only do this in one cpp file
+
+#include "catch2/catch.hpp"
+
+#include <antares/study/parameters.h>
+#include <antares/study/load-options.h>
+
+#include <iostream>
+#include <fstream>
+#include <streambuf>
+
+/* GCC */
+#ifdef __GNUC__
+#if __GNUC__ < 8
+#include <experimental/filesystem>
+namespace fs = std::experimental::filesystem;
+#else
+#include <filesystem>
+namespace fs = std::filesystem;
+#endif
+/* Other compilers */
+#else
+#include <filesystem>
+namespace fs = std::filesystem;
+#endif
+
+/*
+   Small wrapper to ensure temporary files are removed when we no longer need them
+
+   NOTE: This is needed because class Antares::Data::Parameter cannot read
+   from an arbitrary stream, only a file specified by its path. This should be
+   changed.
+*/
+
+class TemporaryFile
+{
+private:
+    const fs::path mPath;
+
+public:
+    TemporaryFile() : mPath(fs::temp_directory_path() / "generaldata.ini")
+    {
+    }
+
+    void write(const std::string& content)
+    {
+        std::ofstream ofs(mPath);
+        // dump content into temporary file
+        ofs << content;
+    }
+
+    std::string read()
+    {
+        std::ifstream in(mPath);
+        if (in)
+        {
+            return (
+              std::string((std::istreambuf_iterator<char>(in)), std::istreambuf_iterator<char>()));
+        }
+        return "error";
+    }
+
+    ~TemporaryFile()
+    {
+        fs::remove(mPath);
+    }
+
+    std::string path() const
+    {
+        return mPath.string();
+    }
+};
+
+namespace {
+  const std::size_t& npos = std::string::npos;
+}
+
+TEST_CASE("Read v800 active years", "[v800]")
+{
+    using namespace Antares::Data;
+    Parameters p;
+    TemporaryFile ini_file;
+    ini_file.write(
+      R"([general]
+mode = Adequacy
+horizon = 2000
+nbyears = 5
+simulation.start = 1
+simulation.end = 365
+january.1st = Monday
+first-month-in-year = january
+first.weekday = Monday
+leapyear = false
+year-by-year = false
+derated = false
+custom-scenario = false
+user-playlist = true
+
+[playlist]
+playlist_reset = false
+playlist_year += 0
+playlist_year += 1)");
+
+    StudyLoadOptions loadOpt;
+    loadOpt.nbYears = 5;
```

Comment:
We could add a test with a `loadOpt.nbyears` different than `nbyears `in `generaldata.ini`

Reply:
what is the expected behavior ?

Reply:
```
/home/florian/Antares_Simulator/src/tests/end-to-end/playlist.cpp:110: FAILED:
  REQUIRE( p.nbYears == 5 )
with expansion:
  3 == 5
```

Reply:
Apparently, `loadOpt.nbYears` overrides `generaldata.ini`'s `nbyears`.

Reply:
I will leave `loadOpt.nbYears=5` for now. I do not know if we want to support other values than the one provided in `generaldata.ini`.